src/general/IWorldEffect.cs

```diff
@@ -37,13 +38,24 @@ public void OnTimePassed(double elapsed, double totalTimePassed)
         {
             var patch = targetWorld.Map.Patches[key];
 
-            foreach (var compound in patch.Biome.Compounds.Keys)
+            // If there are microbes to be eating up the primordial soup, reduce the milk
+            if (patch.SpeciesInPatch.Keys.Count > 0)
             {
-                if (compound.InternalName == "glucose")
+                Compound glucose = null;
+
+                foreach (var compound in patch.Biome.Compounds.Keys)
                 {
-                    var data = patch.Biome.Compounds[compound];
+                    if (compound.InternalName == "glucose")
+                    {
+                        glucose = compound;
+                    }
+                }
 
-                    data.Density *= Constants.GLUCOSE_REDUCTION_RATE;
+                if (glucose != null)
+                {
+                    var toMod = patch.Biome.Compounds[glucose];
```

Comment:
I think this intermediate variable is unnecessary, if this were to be kept the naming would need to be changed as it doesn't follow the style guide.